types/signed_header.go

```diff
@@ -22,7 +26,41 @@ func (sH *SignedHeader) Verify(untrst header.Header) error {
 		// and is a programmer bug
 		panic(fmt.Errorf("%T is not of type %T", untrst, untrstH))
 	}
+	err := untrstH.ValidateBasic()
+	if err != nil {
+		return err
+	}
+	if !bytes.Equal(untrstH.LastHeaderHash[:], sH.Header.Hash()) {
+		return errors.New("Last header hash does not match hash of previous header")
+	}
+	if !bytes.Equal(untrstH.LastCommitHash[:], sH.getLastCommitHash()) {
+		return errors.New("Last header hash does not match hash of previous header")
```

Comment:
copy paste. i assume the error message should reference commit hash?

Reply:
fixed